addons/google_drive/models/google_drive.py

```diff
@@ -15,15 +15,38 @@
 
 from odoo.addons.google_account.models.google_service import GOOGLE_TOKEN_ENDPOINT, TIMEOUT
 
+from datetime import date
+
 _logger = logging.getLogger(__name__)
 
+"""
+Google is depreciating their OOB Auth Flow on 3rd October 2022, the Google Drive
+integration thus become irrelevant after that date.
+
+https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#disallowed-oob
+"""
+GOOGLE_AUTH_DEPRECATION_DATE = date(2022, 10, 3)
+
 
 class GoogleDrive(models.Model):
 
     _name = 'google.drive.config'
     _description = "Google Drive templates config"
 
+    def show_google_deprecation(self):
+        if GOOGLE_AUTH_DEPRECATION_DATE > fields.Date.today():
+            return
+
+        depreciation = _("The Google Drive integration has been deprecated due to changes in Google authentation APIs.")
+        if self.env.is_admin():
+            dummy, action_id = self.env['ir.model.data'].get_object_reference('base_setup', 'action_general_configuration')
+            raise RedirectWarning(depreciation + _(" You can disable the feature in the General Settings"), action_id, _('Go to General Settings'))
```

Comment:
Why not redirect them to the module's form view instead, so they can uninstall from there?

Reply:
Also maybe sending a notification to the admin the first time the code is updated would be a good idea?